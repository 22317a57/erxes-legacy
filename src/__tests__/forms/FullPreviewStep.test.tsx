import { mount, shallow } from 'enzyme';
import FullPreviewStep from 'modules/forms/components/step/FullPreviewStep';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

describe('FullPreviewStep component', () => {
  const defaultProps = {
    type: 'string',
    color: 'red',
    theme: 'default',
    onChange: (name: 'carousel', value: string) => null,
    carousel: 'carousel'
  };

  test('renders shallow successfully', () => {
    shallow(<FullPreviewStep {...defaultProps} />);
  });

  test('renders mount with default props', () => {
    const control = mount(<FullPreviewStep {...defaultProps} />);
    const props = control.props();

    expect(props).toMatchObject(defaultProps);
  });

  test('snapshot matches', () => {
    const rendered = renderer
      .create(<FullPreviewStep {...defaultProps} />)
      .toJSON();

    expect(rendered).toMatchSnapshot();
  });
});
