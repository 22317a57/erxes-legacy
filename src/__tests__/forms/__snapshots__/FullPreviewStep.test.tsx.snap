// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullPreviewStep component snapshot matches 1`] = `
<div
  className="sc-fAJaQT gLizFN"
>
  <div
    className="sc-cNnxps dRaasY"
  >
    <div
      className="sc-jlyJG flDIep"
    >
      <span
        className="active sc-gipzik gocVWe"
        onClick={[Function]}
      >
        <i
          className="icon-laptop sc-bwzfXH hpxbGw"
          icon="laptop"
        />
         
        Desktop
      </span>
      <span
        className="sc-gipzik gocVWe"
        onClick={[Function]}
      >
        <i
          className="icon-paste sc-bwzfXH hpxbGw"
          icon="paste"
        />
         
        Tablet
      </span>
      <span
        className="sc-gipzik gocVWe"
        onClick={[Function]}
      >
        <i
          className="icon-devices sc-bwzfXH hpxbGw"
          icon="devices"
        />
         
        Mobile
      </span>
    </div>
    <div
      className="sc-dPNhBE eOpeLY"
    >
      <div
        className="sc-aewfc dsStAJ"
      >
        <div
          className="sc-isBZXS htStnM"
        >
          <div
            className="sc-OxbzP jocSpk"
            style={
              Object {
                "backgroundColor": "default",
              }
            }
          >
            <div />
          </div>
          <div
            className="sc-lnrBVv jitmGX"
          >
            <div
              className="sc-fvLVrH kjchyg"
            >
              <div
                className="sc-dBAPYN igvoAR"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <ul
      className="sc-bJTOcE foaIyr"
    >
      <div
        className="sc-geAPOV hJwMxA"
        selected={false}
      >
        <li>
          <span
            onClick={[Function]}
          />
        </li>
        <span>
          CallOut
        </span>
      </div>
      <div
        className="sc-geAPOV hJwMxA"
        selected={false}
      >
        <li>
          <span
            onClick={[Function]}
          />
        </li>
        <span>
          Form
        </span>
      </div>
      <div
        className="sc-geAPOV hJwMxA"
        selected={false}
      >
        <li>
          <span
            onClick={[Function]}
          />
        </li>
        <span>
          Success
        </span>
      </div>
    </ul>
  </div>
</div>
`;
